ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/sync/worldstate/TrieNodeDataRequest.java

```diff
@@ -58,7 +59,11 @@
   private Stream<NodeDataRequest> getRequestsFromChildTrieNode(final Node<BytesValue> trieNode) {
     if (trieNode.isReferencedByHash()) {
       // If child nodes are reference by hash, we need to download them
-      NodeDataRequest req = createChildNodeDataRequest(Hash.wrap(trieNode.getHash()));
+      final Hash hash = Hash.wrap(trieNode.getHash());
+      if (hash.equals(MerklePatriciaTrie.EMPTY_TRIE_NODE_HASH) || hash.equals(BytesValue.EMPTY)) {
```

Comment:
consider inverting the ".equals(...) statements so that you never have to worry about the hash being null.

Reply:
We literally just created the hash on the line above. It can't possibly be null.

Reply:
for constants it is a solid pattern.  even if in this case it is impossible (until hash.wrap starts returning nulls)

Reply:
If we're requesting data by hash with a null hash we're screwed *long* before here.

Reply:
`CONSTANT.equals(variable)` is a pattern that keeps us from thinking "have I initialized `variable`" While in this one case it may be true we shouldn't have to think about it when we use the constant first pattern.  It's also a rule for Google's internal Java readability process (separate form the code formatting guides).

Reply:
I'm aware of the pattern but not having to speak like Yoda is really quite nice when there's absolutely no way it can possibly be null.  But just to get out of this discussion I've flipped it round. :)